pkg/kn/core/root.go

```diff
@@ -39,7 +36,10 @@ func NewKnCommand(params ...commands.KnParams) *cobra.Command {
 	} else {
 		panic("Too many params objects to NewKnCommand")
 	}
-	p.Initialize()
+	err := p.Initialize()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to initialize command %v", err))
```

Comment:
Wondering if `panic` is the right thing to do here? I can understand the rational but the error to user will be kind of ugly. Returning error with message will be handled like all errors in `main.go`... 

Though I can also see the argument that this is not a common error and panicking is warranted. Conflicted a bit.

Reply:
+1 for not panicing but exiting with a "normal" error. I think it's not impossible that initialization fails, e.g. when some .kube config file is not readable. We always strive for nice error messages.